src/doc/style-guide/src/statements.md

```diff
@@ -101,22 +101,69 @@ let Foo {
 
 #### else blocks (let-else statements)
 
-If a let statement contains an `else` component, also known as a let-else statement,
-then the `else` component should be formatted according to the same rules as the `else` block
-in [control flow expressions (i.e. if-else, and if-let-else expressions)](./expressions.md#control-flow-expressions).
-Apply the same formatting rules to the components preceding
-the `else` block (i.e. the `let pattern: Type = initializer_expr ...` portion)
-as described [above](#let-statements)
```

Comment:
I forget where, but I know there's been some recent discussion around trying to encapsulate certain prescriptions in one place to avoid duplication, and I'd add, the potential for accidental drift (e.g. breaks around the assignment operator).

I completely agree with that goal, and while it's likely best punted to 2024 edition work, I feel like the basics of "how to write an else" block are something we could capture in one place and then reuse (e.g. the `else` keyword must be followed by a space before the opening brace, if multilined then the closing brace must be preceded by a break, etc.)

Then any syntax specific context can respectively specify where the `else` keyword should start, and any unique wrapping rules